examples/c/Makefile

```diff
@@ -107,10 +106,10 @@ $(LIBBLAZESYM_HEADER): $(LIBBLAZESYM_SRC)/target/release/libblazesym.a | $(OUTPU
 	$(Q)cp $(LIBBLAZESYM_SRC)/target/release/blazesym.h $@
 
 # Build BPF code
-$(OUTPUT)/%.bpf.o: %.bpf.c $(LIBBPF_OBJ) $(wildcard %.h) $(VMLINUX) | $(OUTPUT)
+$(OUTPUT)/%.bpf.o: %.bpf.c $(LIBBPF_OBJ) $(wildcard %.h) $(VMLINUX) | $(OUTPUT) $(BPFTOOL)
 	$(call msg,BPF,$@)
-	$(Q)$(CLANG) -g -O2 -target bpf -D__TARGET_ARCH_$(ARCH) $(INCLUDES) $(CLANG_BPF_SYS_INCLUDES) -c $(filter %.c,$^) -o $@
-	$(Q)$(LLVM_STRIP) -g $@ # strip useless DWARF info
+	$(Q)$(CLANG) -g -O2 -target bpf -D__TARGET_ARCH_$(ARCH) $(INCLUDES) $(CLANG_BPF_SYS_INCLUDES) -c $(filter %.c,$^) -o $(@D)/$*.tmp.bpf.o
+	$(Q)$(BPFTOOL) gen object $@ $(@D)/$*.tmp.bpf.o
```

Comment:
how about replacing "$(@D)/$*.tmp.bpf.o" with "$($@:.bpf.o=.tmp.bpf.o)". Seems cleaner and more straightforward?

Reply:
@anakryiko, thank you for suggesting a better alternative. I'm facing an issue though.
I found out from the [doc](https://makefiletutorial.com/#string-substitution) that the expression you suggested is shorthand for the `patsubst` function call.
In our case, `$(patsubst %.bpf.o,%.tmp.bpf.o,$@)` is working fine but not `$($@:.bpf.o=.tmp.bpf.o)` (returns empty value). I'm not sure why! Any thoughts?

Reply:
ChatGPT says it's how make works :) let's use `$(patsubst)`, no big deal!

Reply:
my main concern is the use of `$(@D)/$*` which are less frequently used and less familiar expressions, so let's do patsubst with `$@`